Automattic/wp-calypso
# FHT: Hosting trial appears in the plans-new-hosted-site plans grid

Closes https://github.com/Automattic/dotcom-forge/issues/4008.

## Proposed Changes

* Adds the free trial to the plans grid during the `/setup/new-hosted-site/plans` step
* Plan gets added to the site and transferred to Atomic
* User is redirected to `/home/%s`

<img width="1191" alt="image" src="https://github.com/Automattic/wp-calypso/assets/26530524/f4affb96-5a9d-46a8-957d-05d93d56f85e">

## Testing Instructions

* Confirm that we aren't inadvertently adding the trial plan to the plans grid in any unexpected locations! e.g. the trial plan appearing during the default `/start` flow.

## Known issues

- there is no `$0` (this should be displayed after we add the trial plan to the `/plans` endpoint)
- the Atomic transfer is not yet happening, this will be addressed in https://github.com/Automattic/dotcom-forge/issues/4046#top. The network request says that no Atomic request was initiated
